src/main/drivers/rx_nrf24l01.c

```diff
@@ -0,0 +1,346 @@
+    DISABLE_NRF24();
+    return true;
+}
+
+uint8_t NRF24L01_ReadReg(uint8_t reg)
+{
+    ENABLE_NRF24();
+    spiTransferByte(NRF24_SPI_INSTANCE, R_REGISTER | (REGISTER_MASK & reg));
+    uint8_t data = spiTransferByte(NRF24_SPI_INSTANCE, 0xFF);
+    DISABLE_NRF24();
+    return data;
+}
+
+uint8_t NRF24L01_WriteRegisterMulti(uint8_t reg, const uint8_t data[], uint8_t length)
+{
+    ENABLE_NRF24();
+    uint8_t res = spiTransferByte(NRF24_SPI_INSTANCE, W_REGISTER | ( REGISTER_MASK & reg));
+    for (uint8_t i = 0; i < length; i++)
+    {
+        spiTransferByte(NRF24_SPI_INSTANCE, data[i]);
+    }
+    DISABLE_NRF24();
+    return res;
+}
+
+uint8_t NRF24L01_WritePayload(const uint8_t *data, uint8_t length)
+{
+    ENABLE_NRF24();
+    uint8_t res = spiTransferByte(NRF24_SPI_INSTANCE, W_TX_PAYLOAD);
+    for (uint8_t i = 0; i < length; i++)
+    {
+        spiTransferByte(NRF24_SPI_INSTANCE, data[i]);
+    }
+    DISABLE_NRF24();
+    return res;
+}
+
+uint8_t NRF24L01_ReadRegisterMulti(uint8_t reg, uint8_t data[], uint8_t length)
+{
+    ENABLE_NRF24();
+    uint8_t res = spiTransferByte(NRF24_SPI_INSTANCE, R_REGISTER | (REGISTER_MASK & reg));
+    for(uint8_t i = 0; i < length; i++)
+    {
+        data[i] = spiTransferByte(NRF24_SPI_INSTANCE, 0xFF);
+    }
+    DISABLE_NRF24();
+    return res;
+}
+
+uint8_t NRF24L01_ReadPayload(uint8_t *data, uint8_t length)
+{
+    ENABLE_NRF24();
+    uint8_t res = spiTransferByte(NRF24_SPI_INSTANCE, R_RX_PAYLOAD);
+    for(uint8_t i = 0; i < length; i++)
+    {
+        data[i] = spiTransferByte(NRF24_SPI_INSTANCE, 0xFF);
+    }
+    DISABLE_NRF24();
+    return res;
+}
+
+void NRF24L01_FlushTx()
+{
+    ENABLE_NRF24();
+    spiTransferByte(NRF24_SPI_INSTANCE, FLUSH_TX);
+    DISABLE_NRF24();
+}
+
+void NRF24L01_FlushRx()
+{
+    ENABLE_NRF24();
+    spiTransferByte(NRF24_SPI_INSTANCE, FLUSH_RX);
+    DISABLE_NRF24();
+}
+
+void NRF24L01_Activate(uint8_t code)
+{
+    ENABLE_NRF24();
+    spiTransferByte(NRF24_SPI_INSTANCE, ACTIVATE);
+    spiTransferByte(NRF24_SPI_INSTANCE, code);
+    DISABLE_NRF24();
+}
+#endif // UNIT_TEST
+
+uint8_t NRF24L01_SetBitrate(uint8_t bitrate)
+{
+    // Note that bitrate 250kbps (and bit RF_DR_LOW) is valid only
+    // for nRF24L01+. There is no way to programmatically tell it from
+    // older version, nRF24L01, but the older is practically phased out
+    // by Nordic, so we assume that we deal with with modern version.
+
+    // Bit 0 goes to RF_DR_HIGH, bit 1 - to RF_DR_LOW
+    rf_setup = (rf_setup & 0xD7) | ((bitrate & 0x02) << 4) | ((bitrate & 0x01) << 3);
+    return NRF24L01_WriteReg(NRF24L01_06_RF_SETUP, rf_setup);
+}
+
+uint8_t NRF24L01_SetPower(uint8_t power)
+{
+    if(power > 3)
+        power = 3;
+    // Power is in range 0..3 for nRF24L01
+    rf_setup = (rf_setup & 0xF9) | ((power & 0x03) << 1);
+    return NRF24L01_WriteReg(NRF24L01_06_RF_SETUP, rf_setup);
+}
+
+void NRF24L01_SetTxRxMode(enum TXRX_State mode)
```

Comment:
Only RX mode is used now and initialized only once .. maybe remove problematic code below and support only one code path